integration-tests/tests/src/tests/dynamic-schema-updates.ts

```diff
@@ -17,41 +17,38 @@
 ////////////////////////////////////////////////////////////////////////////
 
 import { expect } from "chai";
-import Realm from "realm";
+import { openRealmBeforeEach } from "../hooks";
 
 import { PersonSchema, DogSchema } from "../schemas/person-and-dogs";
 
 describe("realm._updateSchema", () => {
-  beforeEach(() => {
-    Realm.clearTestState();
+  openRealmBeforeEach({
+    schema: [PersonSchema, DogSchema],
   });
 
-  it("is a function", () => {
-    const realm = new Realm({ schema: [PersonSchema, DogSchema] });
-    expect(realm.schema).to.be.an("array");
+  it("is a function", function (this: RealmContext) {
+    expect(this.realm.schema).to.be.an("array");
     // There is a function defined on the Realm
-    expect(realm._updateSchema).to.be.a("function");
+    expect(this.realm._updateSchema).to.be.a("function");
     // Expect no enumerable field on the schema property
-    expect(Object.keys(realm.schema)).to.not.contain("update");
+    expect(Object.keys(this.realm.schema)).to.not.contain("update");
     // TODO: This function gets put on the schema to
-    // expect(realm.schema.update).to.be.a("function");
+    // expect(this.realm.schema.update).to.be.a("function");
   });
 
-  it("creates a class schema from a name", () => {
-    const realm = new Realm({ schema: [PersonSchema, DogSchema] });
-    realm.write(() => {
-      realm._updateSchema([...realm.schema, { name: "MyClass", properties: {} }]);
+  it("creates a class schema from a name", function (this: RealmContext) {
+    this.realm.write(() => {
+      this.realm._updateSchema([...this.realm.schema, { name: "MyClass", properties: {} }]);
     });
-    const classNames = realm.schema.map((s) => s.name);
+    const classNames = this.realm.schema.map((s) => s.name);
```

Comment:
I wonder if it would be worth adding a note in the docs on how to use `this` (`this.realm`, `this.config`, gotchas to be aware of e.g. you can only access them in a `it` block, you must use `function` not `() => {}`, you should type the `this` argument) at some point